presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -78,19 +78,25 @@ public FileBasedAccessControl(FileBasedAccessControlConfig config)
     @Override
     public void checkCanCreateSchema(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, String schemaName)
     {
-        denyCreateSchema(schemaName);
+        if (!isSchemaOwner(identity, schemaName)) {
```

Comment:
So the idea is you can create any schema you would become an owner of?  IIRC you normally need to be an owner of the catalog to create or rename schemas.  I think this is a reasonable hack for the file connector, but it should be documented somewhere.

@martint or @electrum thoughts?

Reply:
This seems reasonable to me. I had originally suggested a separate permission for schema creation, but it doesn't make sense to allow creating a schema for which you would have no ownership.

Reply:
I think the same logic should apply to table and view creation: rather than checking schema ownership, check whether the user owns the table.

Reply:
That seems reasonable also.  I'd prefer for it to be consistent.

Reply:
Isn’t this a chicken-and-egg issue? A user can’t be owner of a schema that does not yet exist... what am I missing?

Reply:
@martint the permissions file allows you to have permissions for anything regardless of its existence or not.  Since file based security system is totally generic, there isn't a good way to restrict that even if you wanted to, and instead we rely on the knowledge that the engine only checks for stuff that exists.

Reply:
> I think the same logic should apply to table and view creation: rather than checking schema ownership, check whether the user owns the table.

@electrum this sounds reasonable. I am concerned this may reduce protection in some cases though. Let's do it separately.